.github/workflows/build-check.yaml

```diff
@@ -0,0 +1,32 @@
+name: Build
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        arch: [x86_64, aarch64, loongarch64, powerpc64le]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Install Zig compiler
+        uses: mlugg/setup-zig@v1
+      - name: Install Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '~1.24.2'
+      - name: Install make on Windows
+        if: matrix.os == 'windows-latest'
```

Comment:
@ifraixedes I think we need to somehow skip Windows builds on `loongarch64` and `powerpc64le`. I don't think `zig` will support those, will it?

I've also removed support for `powerpcle64` in my change. Should I add it back?